crates/ruma-client-api/src/error.rs

```diff
@@ -230,6 +230,31 @@ impl fmt::Display for Error {
 
 impl std::error::Error for Error {}
 
+impl From<FromHttpRequestError> for Error {
+    fn from(error: FromHttpRequestError) -> Self {
+        let kind = match error {
+            FromHttpRequestError::Deserialization(DeserializationError::Json(t)) => {
+                match t.is_syntax() {
+                    true => ErrorKind::NotJson,
+                    false => ErrorKind::BadJson,
+                }
+            }
+            FromHttpRequestError::Deserialization(_anything) => ErrorKind::InvalidParam,
+            _ => ErrorKind::Unknown,
+        };
+        Self {
+            message: match &kind {
+                ErrorKind::NotJson => "The JSON is malformed and cannot be parsed".into(),
+                ErrorKind::BadJson => "JSON was parsed but failed to deserialize properly".into(),
+                ErrorKind::InvalidParam => "There was one or more invalid parameters".into(),
+                _ => "Not sure what error has occured".into(),
+            },
+            kind,
+            status_code: http::StatusCode::from_u16(400_u16).unwrap(),
```

Comment:
Not all errors are client errors. An IO error would be an example of something that more like be a server-side error. Also, `StatusCode` has associated constants that can be used instead of requiring `.unwrap()` here.

```suggestion
            status_code: http::StatusCode::BAD_REQUEST,
```